src/urllib3/connectionpool.py

```diff
@@ -848,26 +871,26 @@ class HTTPSConnectionPool(HTTPConnectionPool):
 
     def __init__(
         self,
-        host,
-        port=None,
-        timeout=Timeout.DEFAULT_TIMEOUT,
-        maxsize=1,
-        block=False,
-        headers=None,
-        retries=None,
-        _proxy=None,
-        _proxy_headers=None,
-        key_file=None,
-        cert_file=None,
-        cert_reqs=None,
-        key_password=None,
-        ca_certs=None,
-        ssl_version=None,
-        assert_hostname=None,
-        assert_fingerprint=None,
-        ca_cert_dir=None,
-        **conn_kw,
-    ):
+        host: Optional[str],
+        port: Optional[int] = None,
+        timeout: Union[Timeout, int, float, object] = Timeout.DEFAULT_TIMEOUT,
+        maxsize: int = 1,
+        block: bool = False,
+        headers: Optional[Mapping[str, str]] = None,
+        retries: Optional[Union[Retry, bool, int]] = None,
+        _proxy: Optional[Url] = None,
+        _proxy_headers: Optional[Mapping[str, str]] = None,
+        key_file: Optional[str] = None,
+        cert_file: Optional[str] = None,
+        cert_reqs: Optional[int] = None,
+        key_password: Optional[str] = None,
+        ca_certs: Optional[str] = None,
+        ssl_version: Optional[int] = None,
```

Comment:
`Union[str, int]`?

Reply:
Done

Reply:
after this change we got:

`src/urllib3/connectionpool.py:938: error: Incompatible types in assignment (expression has type "Union[int, str, None]", variable has type "Optional[int]")`